### PR TITLE
Remove default full report when requiring a single report.

### PR DESCRIPTION
**Pull request by icharlie**
Remove the default full report since it is a single report.

### CODE REVIEW

File: CodeSniffer/CLI.php [hunk 1]
```diff
@@ -663,6 +663,7 @@ public function processLongArgument($arg, $pos)
                     }//end if
                 } else {
                     // This is a single report.
+                    unset($this->values['reports'['full']])
```
Comment (aik099) [inline CodeSniffer/CLI.php]: This results in syntax error on PHP 5.3.

Comment (icharlie) [inline CodeSniffer/CLI.php]: My mistake. Thanks!

**Comment by gsherwood** [comment]
I don't know what this change is designed to fix and it actually breaks a pretty key feature of PHPCS, which is the ability to print multiple reports at a time. For example, this change breaks this command:

`phpcs --report=full --report=summary /path/to/code`

But because of the ordering, would still allow:

`phpcs --report=summary --report=full /path/to/code`

PHPCS doesn't hard-coded a default report format before processing the CLI argument (it uses the full report only if no reports were specified) so maybe your full report is being set via a config setting or inside a custom ruleset? If this is the case, you should probably remove it from there as it would be forcing that report format at all times.

**Comment by icharlie** [comment]
@gsherwood Thanks! After, I use `phpcs --config-delete` and it works correctly. But I can't find `ruleset.xml` or `phpcs.xml`. How can I find the configuration file used by phpcs?

**Comment by gsherwood** [comment]
> How can I find the configuration file used by phpcs?

Run the `phpcs` command with the `-vv` command line argument and the first chunk of output will show you how it is setting all its rules. Run it on a small file though because the `-vv` output can be pretty long.

**Comment by icharlie** [comment]
@gsherwood Thank you!!

**Pull request closed without merging**